go/arrow/csv/reader.go

```diff
@@ -48,6 +48,8 @@ type Reader struct {
 
 	header bool
 	once   sync.Once
+
+	null string
```

Comment:
thinking out loud but it seems that, by using `string` and its value to decide whether a value is Null or not might introduce a hysteresis effect while reading columns of strings.
it may very well happen that for a row one gets an empty string that we'd identify as Null (if the reader is configured with `""` as the Null value. I guess that's why you put `"NULL"` as a default value.)
I agree: it's a very degenerate case.

this could be fixed by using a `*string` (or a pair `bool`,`string` if we're worried about memory locality).

what's your take on this issue?
(I am not saying we should go the `*string` route: this has to be weighted against the probability/likelihood of this hysteresis effect to happen)

Reply:
Hi @sbinet - I'm not sure I understand what problem you're trying to solve. Could you clarify with an example?

Reply:
it was more of a gedanken experiment than a real use case.

trying to come up with an example made me realize what I was handwaving wasn't completely correct.

so let's just forget my earlier issue with the empty string `""` and consider the following.

if you had this CSV file, a listing of tools:
```
id,vendor,product
42,NULL,sonic screwdriver
```

for some reason, the valid name for the vendor is the string `"NULL"`.
even if the `arrow/csv.Reader` wasn't configured with `csv.WithNullString("NULL")`, the value we'll get out of `arrow/csv.NewReader` will be a `arrow/csv.Reader` with a field `r.null == "NULL"`.

If I am not mistaken, reading the second column with that `Reader` will return an array whose element is marked `null`.

I guess what I am getting at, is that there doesn't seem to be a way to disable the `null` handling.

this could be left as is, but this would need to be documented (especially the default "null" string)

Reply:
Ah, of course. Thanks.

I read through the C++ implementation to see what it does. For string and binary columns, it has a separate bool strings_can_be_null, which defaults to false. Seems reasonable.

From a performance perspective, for string or binary columns the C++ implementation also hoists the conditional out of the inner loop by using a visitor pattern with a specialized implementation based on whether strings_can_be_null is set. For numeric columns it seems to be roughly the same as what I have here -- always testing for null strings on a per-field/per-row basis. 

Following your earlier question, I haven't yet performance-tested this. I'll add the off-by-default and do some performance evaluation of a couple different implementations.

Reply:
thanks :)